src/components/CodeOverview/index.tsx

```diff
@@ -0,0 +1,217 @@
+import * as React from 'react';
+import { withRouter, Link, RouteComponentProps } from 'react-router-dom';
+import chunk from 'lodash.chunk';
+import debounce from 'lodash.debounce';
+
+import { getCodeLineAnchor, getLines } from '../CodeView/utils';
+import styles from './styles.module.scss';
+import { gettext } from '../../utils';
+import { Version } from '../../reducers/versions';
+import LinterProvider, { LinterProviderInfo } from '../LinterProvider';
+import CodeLineShapes from '../CodeLineShapes';
+import {
+  AllLineShapes,
+  LineShapes,
+  generateLineShapes,
+} from '../CodeLineShapes/utils';
+
+export type PublicProps = {
+  content: string;
+  version: Version;
+};
+
+export type DefaultProps = {
+  _debounce: typeof debounce;
+  _window: {
+    addEventListener: typeof window.addEventListener;
+    removeEventListener: typeof window.removeEventListener;
+  };
+  overviewPadding: number;
+  rowTopPadding: number;
+  rowHeight: number;
+};
+
+export type Props = PublicProps & DefaultProps & RouteComponentProps;
+
+type State = {
+  // This is the height of the overview div in pixels.
+  overviewHeight: number | null;
+};
+
+export class CodeOverviewBase extends React.Component<Props, State> {
+  static defaultProps = {
+    _debounce: debounce,
+    _window: window,
+    // This is the padding of the overview container.
+    overviewPadding: 10,
+    rowTopPadding: 2,
+    // This is the height of the row, including rowTopPadding.
+    rowHeight: 10,
+  };
+
+  public state = { overviewHeight: null };
```

Comment:
`overviewHeight` defaulting to `null` does not seem to be explicitly covered via a test. If I change the value to something other than `null` then the test for `fitLineShapesIntoOverview -> requires overviewHeight` fails, but that's really testing something different.

Do you think we should have a test that asserts that `overviewHeight` defaults to `null`, or does that seem unnecessary?

Reply:
Ah, there was this test https://github.com/mozilla/addons-code-manager/blob/2bf0c9f8c50cfe8a441ef1152c631fb43bed8751/src/components/CodeOverview/index.spec.tsx#L187

...but it was using `setState()` which isn't so helpful. I fixed it.